src/presentations/part-1/assets/examples/go/nix/package.nix

```diff
@@ -0,0 +1,14 @@
+{pkgs, ...}: let
```

Comment:
suggestion: note sure: 

we could use `pkgs.callPackage ./package.nix` which is the standard way of "calling" the build function in `nixpkgs`.  So the `nix/package/executable.nix` takes all inputs needed to build the packages from a package set `pkgs`.

```nix
{lib, buildGoModule}: buildGoModule ...
```
and you assing it to the `default` in `flake.nix` by `forAllSystems {pkgs, ...}: { default = pkgs.callPackage ./nix/package/executable.nix}`

Reply:
pkgs.callPackages uses [`builtins.](https://noogle.dev/f/builtins/functionArgs) to detect the arguments and then injects those from `pkgs` etc.

Reply:
Leaviing this unresolved as I don't really understand the benefit atm